### ISSUE TITLE
SDK crashes trying to connect terminal due to a NPE

### ISSUE THREAD

**Issue by jorge-farside** [stripe/stripe-terminal-android#196]
## Summary
Got this from firebase:
```
Fatal Exception: java.lang.NullPointerException: result.device.name must not be null
       at com.stripe.core.bbpos.BBPOSBluetoothDiscoveryController$discover$1.onScanResult(BBPOSBluetoothDiscoveryController.kt:45)
       at android.bluetooth.le.BluetoothLeScanner$BleScanCallbackWrapper$1.run(BluetoothLeScanner.java:486)
       at android.os.Handler.handleCallback(Handler.java:883)
       at android.os.Handler.dispatchMessage(Handler.java:100)
       at android.os.Looper.loop(Looper.java:214)
       at android.app.ActivityThread.main(ActivityThread.java:7386)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:980)
```

## Android version
Android 10

## Impacted devices (Android devices or readers)
Chipper

## SDK version
implementation 'com.stripe:stripe-android:16.8.0'
implementation "com.stripe:stripeterminal:2.5.1"

**Comment by dhenry-stripe** [comment]
Hi @jorge-farside, the exception is being thrown because Android SDK's `BluetoothDevice.getName()` is returning null. The implementation[0] provides more details as to the possible causes. We'll remove the `NullPointerException` in the next release. You'll need to evaluate things on your side too though to ensure that the Android SDK can retrieve the bluetooth device name (perhaps bluetooth is getting disabled?[1]).

[0] https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/bluetooth/BluetoothDevice.java#1271 
[1] https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/bluetooth/BluetoothDevice.java#1279

**Comment by dhenry-stripe** [comment]
Resolved in [v2.6.0](https://github.com/stripe/stripe-terminal-android/releases/tag/v2.6.0).
